sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataCreateUpdateDeleteTest.java

```diff
@@ -0,0 +1,495 @@
+package com.microsoft.appcenter.data;
+
+import com.microsoft.appcenter.data.client.CosmosDb;
+import com.microsoft.appcenter.data.client.TokenExchange;
+import com.microsoft.appcenter.data.exception.DataException;
+import com.microsoft.appcenter.data.models.DocumentWrapper;
+import com.microsoft.appcenter.data.models.LocalDocument;
+import com.microsoft.appcenter.data.models.ReadOptions;
+import com.microsoft.appcenter.data.models.TokenResult;
+import com.microsoft.appcenter.data.models.WriteOptions;
+import com.microsoft.appcenter.http.HttpClient;
+import com.microsoft.appcenter.http.HttpException;
+import com.microsoft.appcenter.http.ServiceCall;
+import com.microsoft.appcenter.http.ServiceCallback;
+import com.microsoft.appcenter.utils.async.AppCenterFuture;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+import org.hamcrest.CoreMatchers;
+import org.json.JSONException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+import javax.net.ssl.SSLException;
+
+import static com.microsoft.appcenter.data.Constants.PENDING_OPERATION_CREATE_VALUE;
+import static com.microsoft.appcenter.data.Constants.PENDING_OPERATION_REPLACE_VALUE;
+import static com.microsoft.appcenter.data.Constants.PREFERENCE_PARTITION_PREFIX;
+import static com.microsoft.appcenter.data.DefaultPartitions.USER_DOCUMENTS;
+import static com.microsoft.appcenter.http.DefaultHttpClient.METHOD_DELETE;
+import static com.microsoft.appcenter.http.DefaultHttpClient.METHOD_POST;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Matchers.notNull;
+import static org.mockito.Matchers.refEq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyNoMoreInteractions;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({
```

Comment:
Can this be moved to Abstract class without side effect? Seems like duplicated in all children.